webapp/src/main.rs

```diff
@@ -82,12 +86,23 @@ async fn main() -> Result<()> {
     let (rx, tx) = AppSubscribers::new().await;
     native::event::subscribe(tx);
 
-    serve(using_serve_dir(Arc::new(rx), network), 3001).await?;
+    // configure certificate and private key used by https
+    let config =
+        RustlsConfig::from_pem_file(cert_dir.join("cert.pem"), cert_dir.join("key.pem")).await?;
+
+    let app = router(Arc::new(rx), network);
+
+    // run https server
+    let addr = SocketAddr::from(([0, 0, 0, 0], 3001));
+    tracing::debug!("listening on {}", addr);
+    axum_server::bind_rustls(addr, config)
```

Comment:
I'm still divided whether we should simply run on http and add https using nginx